doc/book/admin/disaster_recovery.rst

```diff
@@ -1,126 +1,162 @@
+
+**Actions:**
+
+You lose a few transactions in the master
+:ref:`write-ahead log file <index-box_persistence>`, which may have not
+transferred to the replica before the crash. If you were able to salvage the master
+``.xlog`` file, you may be able to recover these.
+
+1.  Find out instance UUID from the crashed master :ref:`xlog <internals-wal>`:
+
+    .. code-block:: console
+
+        $ head -5 var/lib/instance001/*.xlog | grep Instance
+        Instance: 9bb111c2-3ff5-36a7-00f4-2b9a573ea660
+
+2.  On the new master, use the UUID to find the position:
+
+    .. code-block:: tarantoolsession
+
+        app:instance002> box.info.vclock[box.space._cluster.index.uuid:select{'9bb111c2-3ff5-36a7-00f4-2b9a573ea660'}[1][1]]
+        ---
+        - 999
+        ...
+
+3.  :ref:`Play the records <tt-play>` from the crashed ``.xlog`` to the new master, starting from the
+    new master position:
+
+    .. code-block:: console
+
+        $ tt play 127.0.0.1:3302 var/lib/instance001/00000000000000000000.xlog \
+                  --from 1000 \
+                  --replica 1 \
+                  --username admin --password secret
 
-          $ tt play <new_master_uri> <xlog_file> --from 23425 --replica 1
 
 .. _admin-disaster_recovery-master_master:
 
---------------------------------------------------------------------------------
 Master-master
---------------------------------------------------------------------------------
+-------------
+
+**Configuration:** :ref:`master-master <replication-bootstrap-master-master>`.
 
-Configuration: Two masters.
+**Problem:** one master has crashed.
 
-Problem: Master#1 has crashed.
+**Actions:**
 
-Your actions:
+1.  Let the load be handled by another master alone.
 
-1. Let the load be handled by master#2 (effective master) alone.
+2.  Remove a crashed master from a replica set.
+
+3.  Set up a replacement for the crashed master on a spare host.
+    Learn more from :ref:`Adding and removing instances <replication-master-master-add-remove-instances>`.
 
-2. Follow the same steps as in the
-:ref:`master-replica <admin-disaster_recovery-master_replica>` recovery scenario
-to create a new master and salvage lost data.
 
 .. _admin-disaster_recovery-data_loss:
 
---------------------------------------------------------------------------------
 Data loss
---------------------------------------------------------------------------------
+---------
+
+**Configuration:** master-replica or master-master.
+
+**Problem:** Data was deleted at one master and this data loss was propagated to the other node (master or replica).
+
+**Actions:**
+
+1.  Put all nodes in read-only mode.
+    Depending on the :ref:`replication.failover <configuration_reference_replication_failover>` mode, this can be done as follows:
+
+    -   ``manual``: change a replica set leader to ``null``.
+    -   ``election``: switch from the ``election`` failover mode to ``manual`` and change a replica set leader to ``null``.
```

Comment:
I wouldn't change the `replication.failover` mode if there is a way solve the problem within the mode. Here we can keep `replication.failover: election` and set `replication.election_mode` to `voter` or `off`.

See details [here](https://github.com/tarantool/tarantool/blob/f6d6e4baf0dbef3333d37653b253f3811b3b36f2/src/box/lua/config/applier/box_cfg.lua#L417-L452).

Reply:
Thanks, fixed.